app/src/main/java/org/mozilla/fenix/search/SearchInteractor.kt

```diff
@@ -0,0 +1,114 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.search
+
+import android.content.Context
+import androidx.navigation.NavController
+import mozilla.components.browser.search.SearchEngine
+import mozilla.components.browser.search.SearchEngineManager
+import mozilla.components.support.ktx.kotlin.isUrl
+import org.mozilla.fenix.BrowserDirection
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.components.metrics.MetricController
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.search.awesomebar.AwesomeBarInteractor
+import org.mozilla.fenix.search.toolbar.ToolbarInteractor
+
+/**
+ * Helper function to get the MetricController off of context.
+ */
+val Context.metrics: MetricController
+    get() = this.components.analytics.metrics
+
+/**
+ * Helper function to get the SearchEngineManager off of context.
+ */
+val Context.searchEngineManager: SearchEngineManager
+    get() = this.components.search.searchEngineManager
+
+/**
+ * Interactor for the search screen
+ * Provides implementations for the AwesomeBarView and ToolbarView
+ */
+class SearchInteractor(
+    private val context: Context,
+    private val navController: NavController,
+    private val store: SearchStore
+) : AwesomeBarInteractor, ToolbarInteractor {
+    override fun onUrlCommitted(url: String) {
+        if (url.isNotBlank()) {
+            (context as HomeActivity).openToBrowserAndLoad(
```

Comment:
Technically, fragments (or the interactors within them) aren't supposed to know this much about their containing activity. But I'll allow it, as we were already doing it. I just don't want us to keep copying this technique for new code methods if we can avoid it.

Reply:
Totally agree. This is definitely on my list of things I want to take a look at this cycle